presto-main/src/test/java/com/facebook/presto/execution/TestMemoryRevokingScheduler.java

```diff
@@ -553,31 +557,47 @@ private void assertMemoryRevokingNotRequested()
         assertMemoryRevokingRequestedFor();
     }
 
-    private SqlTask newSqlTask()
+    private SqlTask newSqlTask(QueryId queryId)
     {
-        TaskId taskId = new TaskId("query", 0, 0, idGeneator.incrementAndGet());
+        QueryContext queryContext = getOrCreateQueryContext(queryId);
+
+        TaskId taskId = new TaskId(queryId.getId(), 0, 0, idGeneator.incrementAndGet());
         URI location = URI.create("fake://task/" + taskId);
 
         return createSqlTask(
                 taskId,
                 location,
                 "fake",
-                new QueryContext(new QueryId("query"),
-                        new DataSize(1, MEGABYTE),
-                        new DataSize(2, MEGABYTE),
-                        new DataSize(1, MEGABYTE),
-                        new DataSize(1, GIGABYTE),
-                        memoryPool,
-                        new TestingGcMonitor(),
-                        executor,
-                        scheduledExecutor,
-                        new DataSize(1, GIGABYTE),
-                        spillSpaceTracker),
+                queryContext,
                 sqlTaskExecutionFactory,
                 new MockExchangeClientSupplier(),
                 executor,
                 Functions.identity(),
                 new DataSize(32, MEGABYTE),
                 new CounterStat());
     }
+
```

Comment:
If the refactors to the test class are applied but the fixes in `MemoryRevokingScheduler` are not (ie: tasks created in tests are assigned to the same real `QueryContext` instance) then almost all tests will fail because of the over-counting tasks within the same query bug.